test/integration/master/master_test.go

```diff
@@ -195,6 +253,59 @@ func TestAutoscalingGroupBackwardCompatibility(t *testing.T) {
 	}
 }
 
+func TestAppsGroupBackwardCompatibility(t *testing.T) {
+	_, s := framework.RunAMaster(nil)
+	defer s.Close()
+	transport := http.DefaultTransport
+
+	requests := []struct {
+		verb                string
+		URL                 string
+		body                string
+		expectedStatusCodes map[int]bool
+		expectedVersion     string
+	}{
+		// Post to extensions endpoint and get back from both: extensions and apps
+		{"POST", extensionsPath("deployments", metav1.NamespaceDefault, ""), deploymentExtensions, integration.Code201, ""},
+		{"GET", extensionsPath("deployments", metav1.NamespaceDefault, "test-deployment"), "", integration.Code200, testapi.Extensions.GroupVersion().String()},
+		{"GET", appsPath("deployments", metav1.NamespaceDefault, "test-deployment"), "", integration.Code200, testapi.Apps.GroupVersion().String()},
+		{"DELETE", extensionsPath("deployments", metav1.NamespaceDefault, "test-deployment"), "", integration.Code200, api.Registry.GroupOrDie(v1.GroupName).GroupVersion.String()},
```

Comment:
I did more tests, and the GET after DELETE will actually return you the object is being still deleted. That's why I'm also forced to use different names for both.